src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -54,13 +54,9 @@ public override void MoveFile(string sourceFullPath, string destFullPath)
         {
             // The desired behavior for Move(source, dest) is to not overwrite the destination file
             // if it exists. Since rename(source, dest) will replace the file at 'dest' if it exists,
-            // a check is added before rename is called. Note that this is a race condition. Possible
-            // better solutions would be to use link/unlink or platform specific functions instead of
-            // rename(source, dest). Regardless of the implementation, an IOException should be thrown if 
-            // 'dest' refers to a file that already exists.
-            if (File.Exists(destFullPath))
-                throw new IOException(SR.GetResourceString(SR.IO_AlreadyExists_Name, destFullPath));
-            while (Interop.libc.rename(sourceFullPath, destFullPath) < 0)
+            // link/unlink are used instead. Note that the Unix FileSystemWatcher will treat a Move 
```

Comment:
If we have a Linux-specific option (ie renameat2, as stated in that bug) then we could simply use NSFileManager on OS X to move the file (or directory). NSFileManager will abort if the item already exists, so it would have the same results as Windows and Linux, see OS X documentation on NSFileManager here: https://developer.apple.com/library/ios/documentation/Cocoa/Reference/Foundation/Classes/NSFileManager_Class/index.html#//apple_ref/occ/instm/NSFileManager/moveItemAtPath:toPath:error:

Would that work? This would prevent different events on different platforms